### PR TITLE
Fix types and add trigger context attributes

### PR DESCRIPTION
**Pull request by bergundy**

### CODE REVIEW

File: lib/functionTemplates/js/binaris.d.ts [hunk 1]
```diff
@@ -1,31 +1,33 @@
 /// <reference types="node" />
-interface BinarisHTTPRequest {
-  env: Record<string, string | undefined>;
-  query: Record<string, string | string[] | undefined>;
+export interface HTTPRequest {
+  env: Record<string, string>;
```
Comment (arielshaqed) [inline lib/functionTemplates/js/binaris.d.ts]: This doesn't match [the type in `bolt`](https://git.io/fjkwF).

File: lib/functionTemplates/js/binaris.d.ts [hunk 2]
```diff
@@ -1,31 +1,33 @@
...
+  query: Record<string, string | string[]>;
```
Comment (arielshaqed) [inline lib/functionTemplates/js/binaris.d.ts]: This also doesn't match [the type in `bolt`](https://git.io/fjkwb).

File: lib/functionTemplates/js/binaris.d.ts [hunk 3]
```diff
@@ -1,31 +1,33 @@
...
   body: Buffer;
   path: string;
   method: string;
   requestId: string;
-  headers: Record<string, string | string[] | undefined>;
+  headers: Record<string, string | string[]>;
```
Comment (arielshaqed) [inline lib/functionTemplates/js/binaris.d.ts]: Ditto.

**Comment by bergundy** [comment]
@arielshaqed PTAL.
Reverted the changes that made it incompatible with bolt.
I think the naming is much better like this.
Agree on adding tests for the types and unifying with bolt's types (#469).

**Pull request merged**